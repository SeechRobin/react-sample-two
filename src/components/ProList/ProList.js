import React, { Component } from "react";
import PropTypes from "prop-types";

import Rating from "../UI/Rating/Rating";

class ProList extends Component {
  render() {
    const { pros } = this.props;
    const headers = ["Id", "Name", "Postcode", "Review Rating"];

    const renderHeader = headers.map(header => {
      return (
        <>
          <span className="pro-list-table-header">
            <strong>{header} </strong>
          </span>
        </>
      );
    });

    const renderTable = pros.map(pro => {
      return (
        <>
          <span key={pro.id}>{pro.id}</span>
          <span>{pro.name}</span>
          <span>{pro.main_address.postcode}</span>
          <span>
            <Rating rating={pro.review_rating} />
          </span>
        </>
      );
    });

    return (
      <>
        <div
          className={
            pros.length > 0 ? "pros-list-table" : "pros-list-table-noresult"
          }
        >
          {pros.length > 0
            ? renderHeader
            : "No results Search for Pros in your area"}
          {renderTable}
        </div>
      </>
    );
  }
}
ProList.propTypes = {
  pros: PropTypes.array
};

export default ProList;
