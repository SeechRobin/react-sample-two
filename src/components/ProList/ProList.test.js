import React from "react";
import { shallow } from "enzyme";

import ProList from "./ProList";

describe("<ProList />", () => {
  let wrapper;

  beforeEach(() => {
    const defaultProps = {
      pros: [
        { id: 1, name: "Pro 1", main_address: "1 Place Ave", review_rating: 4 },
        { id: 2, name: "Pro 2", main_address: "3 Place Ave", review_rating: 5 }
      ]
    };

    wrapper = shallow(<ProList {...defaultProps} />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("clicking submit button", () => {
    wrapper.setProps({ pros: [] });
    expect(wrapper.contains("No results Search for Pros in your area")).toEqual(
      true
    );
  });
});
