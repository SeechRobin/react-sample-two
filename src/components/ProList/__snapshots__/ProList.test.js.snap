// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProList /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProList
    pros={
      Array [
        Object {
          "id": 1,
          "main_address": "1 Place Ave",
          "name": "Pro 1",
          "review_rating": 4,
        },
        Object {
          "id": 2,
          "main_address": "3 Place Ave",
          "name": "Pro 2",
          "review_rating": 5,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="pros-list-table"
      >
        <React.Fragment>
          <span
            className="pro-list-table-header"
          >
            <strong>
              Id
               
            </strong>
          </span>
        </React.Fragment>
        <React.Fragment>
          <span
            className="pro-list-table-header"
          >
            <strong>
              Name
               
            </strong>
          </span>
        </React.Fragment>
        <React.Fragment>
          <span
            className="pro-list-table-header"
          >
            <strong>
              Postcode
               
            </strong>
          </span>
        </React.Fragment>
        <React.Fragment>
          <span
            className="pro-list-table-header"
          >
            <strong>
              Review Rating
               
            </strong>
          </span>
        </React.Fragment>
        <React.Fragment>
          <span>
            1
          </span>
          <span>
            Pro 1
          </span>
          <span />
          <span>
            <Rating
              rating={4}
            />
          </span>
        </React.Fragment>
        <React.Fragment>
          <span>
            2
          </span>
          <span>
            Pro 2
          </span>
          <span />
          <span>
            <Rating
              rating={5}
            />
          </span>
        </React.Fragment>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <React.Fragment>
              <span
                className="pro-list-table-header"
              >
                <strong>
                  Id
                   
                </strong>
              </span>
            </React.Fragment>,
            <React.Fragment>
              <span
                className="pro-list-table-header"
              >
                <strong>
                  Name
                   
                </strong>
              </span>
            </React.Fragment>,
            <React.Fragment>
              <span
                className="pro-list-table-header"
              >
                <strong>
                  Postcode
                   
                </strong>
              </span>
            </React.Fragment>,
            <React.Fragment>
              <span
                className="pro-list-table-header"
              >
                <strong>
                  Review Rating
                   
                </strong>
              </span>
            </React.Fragment>,
          ],
          Array [
            <React.Fragment>
              <span>
                1
              </span>
              <span>
                Pro 1
              </span>
              <span />
              <span>
                <Rating
                  rating={4}
                />
              </span>
            </React.Fragment>,
            <React.Fragment>
              <span>
                2
              </span>
              <span>
                Pro 2
              </span>
              <span />
              <span>
                <Rating
                  rating={5}
                />
              </span>
            </React.Fragment>,
          ],
        ],
        "className": "pros-list-table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="pro-list-table-header"
            >
              <strong>
                Id
                 
              </strong>
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Id
                 
              </strong>,
              "className": "pro-list-table-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Id",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Id",
                " ",
              ],
              "type": "strong",
            },
            "type": "span",
          },
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="pro-list-table-header"
            >
              <strong>
                Name
                 
              </strong>
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Name
                 
              </strong>,
              "className": "pro-list-table-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name",
                " ",
              ],
              "type": "strong",
            },
            "type": "span",
          },
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="pro-list-table-header"
            >
              <strong>
                Postcode
                 
              </strong>
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Postcode
                 
              </strong>,
              "className": "pro-list-table-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Postcode",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Postcode",
                " ",
              ],
              "type": "strong",
            },
            "type": "span",
          },
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="pro-list-table-header"
            >
              <strong>
                Review Rating
                 
              </strong>
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Review Rating
                 
              </strong>,
              "className": "pro-list-table-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Review Rating",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Review Rating",
                " ",
              ],
              "type": "strong",
            },
            "type": "span",
          },
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                1
              </span>,
              <span>
                Pro 1
              </span>,
              <span />,
              <span>
                <Rating
                  rating={4}
                />
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pro 1",
              },
              "ref": null,
              "rendered": "Pro 1",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Rating
                  rating={4}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "rating": 4,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                2
              </span>,
              <span>
                Pro 2
              </span>,
              <span />,
              <span>
                <Rating
                  rating={5}
                />
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": 2,
              },
              "ref": null,
              "rendered": 2,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pro 2",
              },
              "ref": null,
              "rendered": "Pro 2",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Rating
                  rating={5}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "rating": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="pros-list-table"
        >
          <React.Fragment>
            <span
              className="pro-list-table-header"
            >
              <strong>
                Id
                 
              </strong>
            </span>
          </React.Fragment>
          <React.Fragment>
            <span
              className="pro-list-table-header"
            >
              <strong>
                Name
                 
              </strong>
            </span>
          </React.Fragment>
          <React.Fragment>
            <span
              className="pro-list-table-header"
            >
              <strong>
                Postcode
                 
              </strong>
            </span>
          </React.Fragment>
          <React.Fragment>
            <span
              className="pro-list-table-header"
            >
              <strong>
                Review Rating
                 
              </strong>
            </span>
          </React.Fragment>
          <React.Fragment>
            <span>
              1
            </span>
            <span>
              Pro 1
            </span>
            <span />
            <span>
              <Rating
                rating={4}
              />
            </span>
          </React.Fragment>
          <React.Fragment>
            <span>
              2
            </span>
            <span>
              Pro 2
            </span>
            <span />
            <span>
              <Rating
                rating={5}
              />
            </span>
          </React.Fragment>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <React.Fragment>
                <span
                  className="pro-list-table-header"
                >
                  <strong>
                    Id
                     
                  </strong>
                </span>
              </React.Fragment>,
              <React.Fragment>
                <span
                  className="pro-list-table-header"
                >
                  <strong>
                    Name
                     
                  </strong>
                </span>
              </React.Fragment>,
              <React.Fragment>
                <span
                  className="pro-list-table-header"
                >
                  <strong>
                    Postcode
                     
                  </strong>
                </span>
              </React.Fragment>,
              <React.Fragment>
                <span
                  className="pro-list-table-header"
                >
                  <strong>
                    Review Rating
                     
                  </strong>
                </span>
              </React.Fragment>,
            ],
            Array [
              <React.Fragment>
                <span>
                  1
                </span>
                <span>
                  Pro 1
                </span>
                <span />
                <span>
                  <Rating
                    rating={4}
                  />
                </span>
              </React.Fragment>,
              <React.Fragment>
                <span>
                  2
                </span>
                <span>
                  Pro 2
                </span>
                <span />
                <span>
                  <Rating
                    rating={5}
                  />
                </span>
              </React.Fragment>,
            ],
          ],
          "className": "pros-list-table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="pro-list-table-header"
              >
                <strong>
                  Id
                   
                </strong>
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Id
                   
                </strong>,
                "className": "pro-list-table-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Id",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Id",
                  " ",
                ],
                "type": "strong",
              },
              "type": "span",
            },
            "type": Symbol(react.fragment),
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="pro-list-table-header"
              >
                <strong>
                  Name
                   
                </strong>
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Name
                   
                </strong>,
                "className": "pro-list-table-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name",
                  " ",
                ],
                "type": "strong",
              },
              "type": "span",
            },
            "type": Symbol(react.fragment),
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="pro-list-table-header"
              >
                <strong>
                  Postcode
                   
                </strong>
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Postcode
                   
                </strong>,
                "className": "pro-list-table-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Postcode",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Postcode",
                  " ",
                ],
                "type": "strong",
              },
              "type": "span",
            },
            "type": Symbol(react.fragment),
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="pro-list-table-header"
              >
                <strong>
                  Review Rating
                   
                </strong>
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Review Rating
                   
                </strong>,
                "className": "pro-list-table-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Review Rating",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Review Rating",
                  " ",
                ],
                "type": "strong",
              },
              "type": "span",
            },
            "type": Symbol(react.fragment),
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  1
                </span>,
                <span>
                  Pro 1
                </span>,
                <span />,
                <span>
                  <Rating
                    rating={4}
                  />
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pro 1",
                },
                "ref": null,
                "rendered": "Pro 1",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Rating
                    rating={4}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "rating": 4,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
            ],
            "type": Symbol(react.fragment),
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  2
                </span>,
                <span>
                  Pro 2
                </span>,
                <span />,
                <span>
                  <Rating
                    rating={5}
                  />
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": 2,
                },
                "ref": null,
                "rendered": 2,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pro 2",
                },
                "ref": null,
                "rendered": "Pro 2",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Rating
                    rating={5}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "rating": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
            ],
            "type": Symbol(react.fragment),
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
